packages/console/src/pages/ApplicationDetails/components/SettingsFormFields/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { Application } from '@logto/schemas';
+import React from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import CopyToClipboard from '@/components/CopyToClipboard';
+import FormField from '@/components/FormField';
+import MultiTextInput from '@/components/MultiTextInput';
+import TextInput from '@/components/TextInput';
+import useWatchedController from '@/hooks/use-watched-controller';
+import { noSpaceRegex } from '@/utilities/regex';
+
+import * as styles from '../../index.module.scss';
+
+type Props = {
+  authorizationEndpoint: string;
+};
+
+const SettingsFormFields = ({ authorizationEndpoint }: Props) => {
+  const { control, register } = useFormContext<Application>();
+
+  const { value: redirectUris, onChange: onRedirectUriChange } = useWatchedController({
+    control,
+    name: 'oidcClientMetadata.redirectUris',
+  });
+
+  const { value: postSignOutRedirectUris, onChange: onPostSignOutRedirectUriChange } =
+    useWatchedController({
+      control,
+      name: 'oidcClientMetadata.postLogoutRedirectUris',
+    });
+
+  return (
+    <div>
+      <FormField isRequired title="admin_console.application_details.application_name">
+        <TextInput {...register('name', { required: true })} />
+      </FormField>
+      <FormField title="admin_console.application_details.description">
+        <TextInput {...register('description')} />
+      </FormField>
+      <FormField title="admin_console.application_details.authorization_endpoint">
+        <CopyToClipboard className={styles.textField} value={authorizationEndpoint} />
+      </FormField>
+      <FormField isRequired title="admin_console.application_details.redirect_uri">
+        <MultiTextInput
```

Comment:
The usage of this form component looks a bit complicated, ideally, it only need "value" and "onChange" for minimum usage. @logto-io/eng